MapleServer2/Packets/PlayerHostPacket.cs

```diff
@@ -0,0 +1,106 @@
+﻿using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Types;
+
+namespace MapleServer2.Packets
+{
+    public static class PlayerHostPacket
+    {
+        private enum PlayerHostPacketMode : byte
+        {
+            OpenHongbao = 0x0,
+            HongbaoGiftNotice = 0x2,
+            StartMinigame = 0x3,
+            MinigameRewardNotice = 0x4,
+            MinigameRewardReceive = 0x5,
+            AdBalloonWindow = 0x6,
+            AdBalloonPlace = 0x7,
+        }
+        public static Packet OpenHongbao(Player player, HongBao hongBao)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.PLAYER_HOST);
+            pWriter.WriteEnum(PlayerHostPacketMode.OpenHongbao);
+            pWriter.WriteInt(hongBao.ItemId);
+            pWriter.WriteInt(hongBao.Id);
+            pWriter.WriteInt(hongBao.RewardId);
+            pWriter.WriteInt(hongBao.RewardAmount);
+            pWriter.WriteInt(hongBao.MaxReceivers);
+            pWriter.WriteUnicodeString(player.Name);
+            return pWriter;
+        }
+
+        public static Packet HongbaoGiftNotice(Player receiver, HongBao hongBao, int dividedRewardAmount)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.PLAYER_HOST);
+            pWriter.WriteEnum(PlayerHostPacketMode.HongbaoGiftNotice);
+            pWriter.WriteBool(hongBao.Active);
+            switch (hongBao.Active)
+            {
+                case true:
+                    pWriter.WriteInt(hongBao.ItemId);
+                    pWriter.WriteInt(hongBao.RewardId);
+                    pWriter.WriteInt(dividedRewardAmount);
+                    pWriter.WriteUnicodeString(hongBao.Giver.Name);
+                    pWriter.WriteUnicodeString(receiver.Name);
+                    break;
+                case false:
+                    break;
```

Comment:
Do we know for sure that when it's false, there's nothing else to do? if so, this can just be a simple if statement